apps/varnish.go

```diff
@@ -52,3 +53,35 @@ func (r MetricsReceiverVarnish) Pipelines() []otel.Pipeline {
 func init() {
 	confgenerator.MetricsReceiverTypes.RegisterType(func() confgenerator.Component { return &MetricsReceiverVarnish{} })
 }
+
+type LoggingProcessorVarnishAccess struct {
+	confgenerator.ConfigComponent `yaml:",inline"`
+}
+
+func (LoggingProcessorVarnishAccess) Type() string {
+	return "varnish_access"
+}
+
+func (p LoggingProcessorVarnishAccess) Components(tag string, uid string) []fluentbit.Component {
+	return genericAccessLogParser(tag, uid)
```

Comment:
Can we actually use the generic access log parser here? From what I remembered I thought we needed a custom regex rule because varnish logs had fields like `httpRequest.serverIp`?

Reply:
yeah, a generic access log parser can be used. I added the sample line back in if you want to try. The varnishncsa defaults to to [client mode](https://varnish-cache.org/docs/6.0/reference/varnishncsa.html#modes) , so the [varnishncsa log format](https://github.com/varnishcache/varnish-cache/blob/04455d6c3d8b2d810007239cb1cb2b740d7ec8ab/doc/sphinx/reference/varnishncsa.rst#format) reflects the [access log format](https://httpd.apache.org/docs/current/logs.html#accesslog). It was my mistake for putting serverIp since varnishncsa defaults to client mode, not backend mode.